fuel-vm/src/interpreter.rs

```diff
@@ -461,3 +461,27 @@ impl CheckedMetadata for CreateCheckedMetadata {
         self.gas_used_by_predicates = gas_used;
     }
 }
+
+pub(crate) struct TouchedContracts<'vm, I> {
```

Comment:
I think the idea of centralizing the location where we check the list of input contracts is a good abstraction. However, I think `touch` is a bit of a misleading name. In the context of software, `touch` usually means to create a file/modify its timestamp (see https://www.geeksforgeeks.org/touch-command-in-linux-with-examples/). Here, you're using `touch` to panic if the file doesn't exist.
What about calling it something like `InputContracts<..>` and changing `touch` to something like `check`?